### PR TITLE
feat: create embedder&llm for worker based on model type

### PR DESCRIPTION
**Pull request by bjwswang**
#### Which issue(s) this PR fixes

Fixes #268 

#### Special notes for your reviewer

User can use `Embedder` for embedding and `LLM` for llm chat.  Here I defined two kinds of `Provider` which provides the `embedder` and `llm`.

- 3rd party provider,like zhipuai,openai.
- arcadia worker,like our fastchat worker/vllm worker.

For 3rd party provider, the user must provide `endpoint` manually.

For Worker provider ,the user must create a `CR Worker`. Whe worker is ready, our arcadia controller will create `Embedder`/`LLM` based on the Worker's model type. 

- If worker's model type contains `embedding`,then a embedder will be created
- if woker's model type contains `llm`, then a LLM will be created.

### CODE REVIEW

**Comment by bjwswang** [comment]
@CodiumAI-Agent /review

File: api/v1alpha1/worker.go [hunk 1]
```diff
@@ -72,3 +75,45 @@ func (worker Worker) ErrorCondition(msg string) Condition {
 		LastSuccessfulTime: metav1.Now(),
 	}
 }
+
+func (worker Worker) BuildEmbedder() *Embedder {
+	return &Embedder{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: worker.Namespace,
+			Name:      worker.Name + "-worker",
```
Comment (bjwswang) [inline api/v1alpha1/worker.go:83]: Not sure  about this `Name`. Maybe a random string?

Comment (bjwswang) [reply api/v1alpha1/worker.go:83]: @nkwangleiGIT @Abirdcfly @0xff-dev any comments?

File: api/v1alpha1/embedder_types.go [hunk 2]
```diff
@@ -32,8 +32,13 @@ type EmbedderSpec struct {
 	// ServiceType indicates the source type of embedding service
 	ServiceType embeddings.EmbeddingType `json:"serviceType,omitempty"`
 
-	// Enpoint defines connection info
+	// Enpoint defines the connection info
```
Comment (Abirdcfly) [inline api/v1alpha1/embedder_types.go:35]: look like we can have 
```go
type Provider struct {
	Enpoint *Endpoint `json:"endpoint,omitempty"`
	Worker *TypedObjectReference `json:"worker,omitempty"`
}

func (p Provider) Provider() Provider {}
func (p Provider) GetWorker() types.NamespacedName {}
```

then just use it in embedder and llm

Comment (bjwswang) [reply api/v1alpha1/embedder_types.go:35]: You are right. This is much clear.

File: api/v1alpha1/model.go [hunk 3]
```diff
@@ -18,19 +18,35 @@ package v1alpha1
...
+		if strings.ToLower(t) == "llm" {
+			return "true"
+		}
 	}
-	return model.Spec.Types
+	return "false"
+}
+
+// IsLLMModel checks whether this model is a embedding model
+func (model Model) IsEmbeddingModel() string {
```
Comment (Abirdcfly) [inline api/v1alpha1/model.go:43]: maybe return bool?

File: api/v1alpha1/embedder.go [hunk 4]
```diff
@@ -24,6 +24,32 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+func (e Embedder) Provider() Provider {
```
Comment (Abirdcfly) [inline api/v1alpha1/embedder.go:27]: maybe like Model will use IsLLMModel function, Provider should also provider Is3rdParty and IsProviderWorker function?

File: api/v1alpha1/embedder_types.go [hunk 5]
```diff
@@ -32,8 +32,8 @@ type EmbedderSpec struct {
 	// ServiceType indicates the source type of embedding service
 	ServiceType embeddings.EmbeddingType `json:"serviceType,omitempty"`
 
-	// Enpoint defines connection info
-	Enpoint *Endpoint `json:"endpoint,omitempty"`
+	// Provider defines the provider info which provide this embedder service
+	Provider `json:"provider,omitempty"`
```
Comment (Abirdcfly) [inline api/v1alpha1/embedder_types.go:36]: what about use `json:",inline"`?

File: api/v1alpha1/worker.go (hunk 1 above)
Comment (bjwswang) [reply api/v1alpha1/worker.go]: image

File: api/v1alpha1/embedder.go (hunk 4 above)
Comment (bjwswang) [reply api/v1alpha1/embedder.go]: provide type and switch case is good enough

**Review by Abirdcfly — APPROVED** [review]

**Pull request merged**